docs/source/format/DissociatedIPC.rst

```diff
@@ -0,0 +1,335 @@
+      and subsequent addresses would be offsets from this base pointer.
+
+Protocol Description
+====================
+
+There are two possibilities that can occur:
+
+1. The streams of metadata and body data are across separate connections
+
+.. figure:: ./DissociatedIPC/SequenceDiagramSeparate.mmd.svg
+
+2. The streams of metadata and body data are sent simultaneously across the
+   same connection
+
+.. figure:: ./DissociatedIPC/SequenceDiagramSame.mmd.svg
+
+Server Sequence
+---------------
+
+There can be either a single server handling both the IPC Metadata stream and the
+Body data streams, or separate servers for handling the IPC Metadata and the body
+data. This allows for streaming of data across either a single transport pipe or
+two pipes if desired.
+
+Metadata Stream Sequence
+''''''''''''''''''''''''
+
+The standing state of the server is waiting for a **tagged** message with the specified
+``<want_data>`` tag value to initiate a transfer. The body of that message will contain an
+opaque, binary identifier to indicate a particular dataset / data stream to send.
+
+.. note::
+
+  For instance, the **ticket** that was passed with a *FlightInfo* message would be
+  the body of this message. Because it is opaque, it can be anything the server wants
+  to use. The URI and identifier do not need to be given to the client via Flight RPC,
+  but could come across from any transport or protocol desired.
+
+Upon receiving a ``<want_data>`` request, the server *should* respond by sending a stream
+of messages consisting of the following:
+
+* A 5-byte prefix
+  - The first byte of the message indicates the type of message, currently there are only
+    two allowed message types (more types may get added in the future):
+    0) End of Stream
+    1) Flatbuffer IPC Metadata Message
+  - the next 4-bytes are a little-endian, 32-bit integer indicating the sequence number of
+    the message. The first message in the stream (**MUST** always be a schema message) **MUST**
+    have a sequence number of ``0``. Each subsequent message **MUST** increment the number by 
+    ``1``.
+* The full Flatbuffer bytes of an Arrow IPC header
+
+After sending the last metadata message, the server **MUST** indicate the end of the stream
+by sending a message consisting of **exactly** 5 bytes:
+
+* The first byte is ``0``, indicating an **End of Stream** message
+* The last 4 bytes are the sequence number (4-byte, unsigned integer in little-endian byte order)
+
+Data Stream Sequence
+''''''''''''''''''''
+
+If a single server is handling both the data and metadata streams, then the data messages
+**should** begin being sent to the client in parallel with the metadata messages. Otherwise,
+as with the metadata sequence, the standing state of the server is to wait for a **tagged**
+message with the ``<want_data>`` tag value, whose body indicates the dataset / data stream
+to send to the client.
```

Comment:
To make the two cases more similar, do we want separate `want_metadata` and `want_data` tags?

Reply:
I don't think so personally. That would introduce confusion for the scenario where they want a single server to be able to provide both the metadata and body data streams (which currently only requires a single message to initiate both streams) where either we would need to make the protocol more "chatty" (requiring an extra message) in that situation or it would be unclear which (if any) of these tags could be left out for that scenario. 

I also prefer keeping it to a single `want` value, rather than requiring two values.

Reply:
Ok, thanks for the clarification.